json-rpc/src/runtime.rs

```diff
@@ -250,6 +257,7 @@ pub(crate) async fn rpc_endpoint(
     data: Value,
     service: JsonRpcService,
     registry: Arc<RpcRegistry>,
+    user_agent: String,
```

Comment:
Maybe we should look at making this an Option<String> so you don't have to manually create an empty string above?

Reply:
If you want to go down the route of using an Option<String> take a look at `warp::filters::header::optional`